lib/active_merchant/billing/gateways/priority.rb

```diff
@@ -0,0 +1,471 @@
+        # tax1['unitOfMeasure'] = 'EA'
+        # tax1['unitPrice'] = '1.51'
+
+        # params['purchases'] << tax1
+
+        params['reference'] = options['reference']
+        params['replayId'] = nil
+        params['requireSignature'] = false
+        params['reviewIndicator'] = nil
+
+        params['risk'] = options['risk']
+        # risk = params['risk'] = {}
+        # risk['avsAddressMatch'] = false
+        # risk['avsResponse'] = 'No Response from AVS'
+        # risk['avsZipMatch'] = false
+        # risk['cvvMatch'] = true
+        # risk['cvvResponse'] = 'Match'
+        # risk['cvvResponseCode'] = 'M'
+
+        params['settledAmount'] = options['settledAmount']
+        params['settledCurrency'] = options['settledCurrency']
+        params['settledDate'] = options['created']
+        params['shipToCountry'] = options['shipToCountry']
+        params['shouldGetCreditCardLevel'] = options['shouldGetCreditCardLevel']
+        params['source'] = 'Spreedly'
+        params['sourceZip'] = nil
+        params['status'] = options['status']
+        params['tax'] = options['tax']
+        params['taxExempt'] = options['taxExempt']
+        params['tenderType'] = 'Card'
+        params['type'] = options['type']
+      end
+
+      def add_invoice(post, money, options)
+        post[:amount] = amount(money)
+        post[:currency] = (options[:currency] || currency(money))
+      end
+
+      def add_payment(post, payment); end
+
+      def commit(action, params, iid, creditcardnumber, options)
+        response =
+          begin
+            if action == 'void' || action == 'closebatch'
+              ssl_invoke(action, params, iid, creditcardnumber, options)
+            else
+              parse(ssl_invoke(action, params, iid, creditcardnumber, options))
+            end
+          rescue ResponseError => e
+            #   byebug
+            parse(e.response.body)
+          end
+        #   byebug
+        success = success_from(response)
+
+        if response == '' 
+            response = { 'code' => '204' }
+        end
+        Response.new(
+          success,
+          message_from(success, response),
+          response,
+          authorization: success && response['code'] != '204' ? authorization_from(response) : nil,
+          error_code: success || response['code'] == '204' || response == '' ? nil : error_from(response),
+          test: test?
+        )
+      end
+
+      def handle_response(response)
+        #   byebug
+        # return response.body
+        if response.code != '204' && (200...300).cover?(response.code.to_i)
+          response.body
+        else
+
+          if response.code == '204' || response == ''
+            response.body = { 'code' => '204' }
+          else
+            raise ResponseError.new(response)
+          end
+        end
+      end
+
+      def ssl_invoke(action, params, refnumber, creditcardnumber, options)
+        # byebug
+        if action == 'void'
```

Comment:
This could be converted to a switch statement for easier reading

Reply:
Not done

Reply:
@mikebyrneprth thoughts on reason for not wanting to convert to a switch statement?

Reply:
@mikebyrneprth still looking for insight into this ^